### PR TITLE
[WIP] Fixes for RNN tests

### PR DESCRIPTION
**Pull request by tejank10**
Currently, test failures due to `params` getting zero value have been fixed.
What remains to be fixed is: 

- [ ] GRU gradtests
- [ ] CUDNN_BAD_PARAM error

### CODE REVIEW

**Comment by tejank10** [comment]
Thanks for the hint, It's passing all tests now.

File: src/cuda/cudnn.jl [hunk 1]
```diff
@@ -87,11 +87,14 @@ function RNNDesc{T}(mode::Int, input::Int, hidden::Int; layers = 1) where T
     libcudnn_handle[],d[],hidden,layers,dropoutDesc,inputMode,direction,mode,algo,cudnnDataType(T))
 
   w = cuzeros(T, rnnParamSize(T, d[], input))
+  (wx, wh), bias = params(w, input, hidden, ngates(mode))
+  w_ = vcat(wx[:], wh[:], bias)
+  w[1:length(w_)] .= w_
```
Comment (MikeInnes) [inline src/cuda/cudnn.jl]: Are these changes still necessary, given the view fix?

File: src/cuda/cudnn.jl [hunk 2]
```diff
@@ -279,6 +279,9 @@ function RNNDesc(m::CuRNNs{T}) where T
     (m.σ == tanh ? RNN_TANH : RNN_RELU) :
     m isa CuGRU ? GRU : LSTM
   r = RNNDesc{T}(mode, i, h)
+  #w_ = vcat(m.Wi[:], m.Wh[:], m.b)
+  #r.params[1:length(w_)] .= w_
```
Comment (MikeInnes) [inline src/cuda/cudnn.jl]: Can you just clean up the diff here? Otherwise this is good to go.

**Comment by MikeInnes** [comment]
:+1: Thanks!
